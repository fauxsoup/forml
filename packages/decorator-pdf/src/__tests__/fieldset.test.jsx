import FieldSet from "../fieldset";
import Context from "@forml/context";
import React from "react";
import ReactPDF from "@react-pdf/renderer";
import { render } from "@testing-library/react";
import * as decorator from "../";

describe("renders", function () {
  let form;
  let schema;
  let title = "title";
  let description = "description";
  let styles;
  let root;
  let header;
  let content;
  let paper;

  beforeEach(function () {
    schema = { type: "string" };
    form = { type: "fieldset", items: [{ key: [] }] };
  });

  describe("with form options", function () {
    let fields = {
      layout: ["horizontal", "vertical"],
      disableGutters: [true, false],
      disablePadding: [true, false],
      disableMargin: [true, false],
      wrap: [true, false],
      break: [true, false],
    };

    Object.keys(fields).forEach(function (field) {
      fields[field].forEach(function (value) {
        describe(`${field}`, function () {
          test(`${value}`, function () {
            form = { ...form, [field]: value };
            const { container } = render(
              <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                  <FieldSet
                    form={form}
                    title={title}
                    description={description}
                  />
                </Context.Provider>
              </ReactPDF.Page>
            );

            expect(container).toMatchSnapshot();
          });
        });
      });
    });
  });

  test("with title and description", function () {
    const { container } = render(
      <ReactPDF.Page>
        <Context.Provider value={{ decorator }}>
          <FieldSet form={form} title={title} description={description} />
        </Context.Provider>
      </ReactPDF.Page>
    );

    expect(container).toMatchSnapshot();
  });

  test("with title and no description", function () {
    const { container } = render(
      <ReactPDF.Page>
        <Context.Provider value={{ decorator }}>
          <FieldSet form={form} title={title} />
        </Context.Provider>
      </ReactPDF.Page>
    );

    expect(container).toMatchSnapshot();
  });

  test("with description and no title", function () {
    const { container } = render(
      <ReactPDF.Page>
        <Context.Provider value={{ decorator }}>
          <FieldSet form={form} description={title} />
        </Context.Provider>
      </ReactPDF.Page>
    );

    expect(container).toMatchSnapshot();
  });

  test("with no title or description", function () {
    const { container } = render(
      <ReactPDF.Page>
        <Context.Provider value={{ decorator }}>
          <FieldSet form={form} />
        </Context.Provider>
      </ReactPDF.Page>
    );

    expect(container).toMatchSnapshot();
  });

  test("with relevant formSTyles", function () {});
});
