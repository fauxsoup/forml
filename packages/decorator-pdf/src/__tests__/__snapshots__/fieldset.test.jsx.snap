// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with description and no title 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            title
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options break false 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options break true 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options disableGutters false 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options disableGutters true 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 0px 4pt 0px 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options disableMargin false 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options disableMargin true 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 0px; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options disablePadding false 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options disablePadding true 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt 0px 4pt 0px; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options layout horizontal 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: row;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options layout vertical 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options wrap false 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options wrap true 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with no title or description 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1;"
    >
      <view
        style="padding: 4pt; flex-direction: column;"
      />
    </view>
  </page>
</div>
`;

exports[`renders with title and description 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-family: Roboto; font-size: 9pt; font-weight: 200; padding-top: 2pt;"
          >
            description
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with title and no description 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; margin: 4pt; border: 1pt solid black; padding: 0px;"
    >
      <view
        style="display: flex; flex: 1;"
      >
        <view
          style="padding: 4pt; background-color: rgb(238, 238, 238);"
        >
          <text
            style="font-family: Roboto; font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
        </view>
        <view
          style="padding: 4pt; flex-direction: column;"
        />
      </view>
    </view>
  </page>
</div>
`;
